phys2denoise.py

```diff
@@ -0,0 +1,162 @@
+        File containing information for BIDS.
+    """
+    start_time = -time_offset
+    summary = dict(SamplingFrequency=samp_freq,
+                   StartTime=round(start_time, 4),
+                   Columns=ch_name)
+    utils.writejson(outfile, summary, indent=4, sort_keys=False)
+
+
+@due.dcite(
+     Doi(''),
+     path='phys2denoise',
+     description='Creation of regressors for physiological denoising',
+     version=__version__,
+     cite_module=True)
+def phys2denoise(filename, outdir='.', debug=False, quiet=False):
+    """
+    Run main workflow of phys2denoise.
+
+    Runs the parser, does some checks on input, then computes the required metrics.
+
+    Notes
+    -----
+    The code was greatly copied from phys2bids (copyright the physiopy community)
+
+    """
+    # Check options to make them internally coherent pt. I
+    # #!# This can probably be done while parsing?
+    outdir = os.path.abspath(outdir)
+    os.makedirs(outdir)
+    os.makedirs(os.path.join(outdir, 'code'))
+    conversion_path = os.path.join(outdir, 'code', 'conversion')
+    os.makedirs(conversion_path)
+
+    # Create logfile name
+    basename = 'phys2denoise_'
+    extension = 'tsv'
+    isotime = datetime.datetime.now().strftime('%Y-%m-%dT%H%M%S')
+    logname = os.path.join(conversion_path, (basename + isotime + '.' + extension))
+
+    # Set logging format
+    log_formatter = logging.Formatter(
+        '%(asctime)s\t%(name)-12s\t%(levelname)-8s\t%(message)s',
+        datefmt='%Y-%m-%dT%H:%M:%S')
+
+    # Set up logging file and open it for writing
+    log_handler = logging.FileHandler(logname)
+    log_handler.setFormatter(log_formatter)
+    sh = logging.StreamHandler()
+
+    if quiet:
+        logging.basicConfig(level=logging.WARNING,
+                            handlers=[log_handler, sh], format='%(levelname)-10s %(message)s')
+    elif debug:
+        logging.basicConfig(level=logging.DEBUG,
+                            handlers=[log_handler, sh], format='%(levelname)-10s %(message)s')
+    else:
+        logging.basicConfig(level=logging.INFO,
+                            handlers=[log_handler, sh], format='%(levelname)-10s %(message)s')
+
+    version_number = _version.get_versions()['version']
+    LGR.info(f'Currently running phys2denoise version {version_number}')
+    LGR.info(f'Input file is {filename}')
+
+    # Save call.sh
+    arg_str = ' '.join(sys.argv[1:])
+    call_str = f'phys2denoise {arg_str}'
+    f = open(os.path.join(conversion_path, 'call.sh'), "a")
+    f.write(f'#!bin/bash \n{call_str}')
+    f.close()
+
+    # Check options to make them internally coherent pt. II
+    # #!# This can probably be done while parsing?
+    # filename, ftype = utils.check_input_type(filename)
+
+    if not os.path.isfile(filename) and filename is not None:
+        raise FileNotFoundError(f'The file {filename} does not exist!')
+
+    # Read input file
+    phys_in = np.genfromtxt(filename)
+
+
+
+
+
+
+
+
+
```

Comment:
Why so many empty lines?

Reply:
Cause it's not done yet!

Reply:
Make sure you pull from master cause I can still see Travis. 😉

Reply:
Yes, but that is because we didn't merge #12 yet - we should definitely merge that one first!